storagenode/gracefulexit/worker.go

```diff
@@ -91,10 +93,13 @@ func (worker *Worker) Run(ctx context.Context, done func()) (err error) {
 		case *pb.SatelliteMessage_NotReady:
 			break // wait until next worker execution
 		case *pb.SatelliteMessage_TransferPiece:
-			err = worker.transferPiece(ctx, msg.TransferPiece, c)
-			if err != nil {
-				continue
-			}
+			transferPieceMsg := msg.TransferPiece
+			worker.limiter.Go(ctx, func() {
+				err = worker.transferPiece(ctx, transferPieceMsg, c)
+				if err != nil {
+					worker.log.Error("failed to transfer piece.", zap.Stringer("Satellite ID", worker.satelliteID), zap.Error(errs.Wrap(err)))
+				}
+			})
 		case *pb.SatelliteMessage_DeletePiece:
 			pieceID := msg.DeletePiece.OriginalPieceId
 			err := worker.deleteOnePieceOrAll(ctx, &pieceID)
```

Comment:
Any reason not to pass off the delete to the `limiter`?

Reply:
My main reason was that since a deletion involves only local operations, it would be relatively quick compared to a piece transfer. But I don't think it would hurt to add it to the limiter. Do you think we should?

Reply:
I added it so deletions don't cause a bottleneck